static/js/components/widgets/MarkdownEditor.tsx

```diff
@@ -102,15 +101,13 @@ export default function MarkdownEditor(props: Props): JSX.Element {
         },
         toolbar: {
           ...FullEditorConfig.toolbar,
-          items: FullEditorConfig.toolbar.items.filter(item =>
-            hasAttach ?
-              true :
-              item !== ADD_RESOURCE_LINK && item !== ADD_RESOURCE_EMBED
-          )
+          items: FullEditorConfig.toolbar.items
+            .filter(item => link.length !== 0 || item !== ADD_RESOURCE_LINK)
+            .filter(item => embed.length !== 0 || item !== ADD_RESOURCE_EMBED)
```

Comment:
just me - but I find these conditionals confusing - could we say it more plainly instead, something like:

```suggestion
            .filter(item => {
              if (item === ADD_RESOURCE_LINK) {
                return link.length > 0
              }
              if (item === ADD_RESOURCE_EMBED) {
                return embed.length > 0
              }
              return true
            })
```

more of a nitpick really, I just always get tripped up trying to think through boolean conditions and so prefer to write them out as explicitly as possible 🙂

Reply:
Yes that's much cleaner. I dunno why my brain was having a hard time putting this in one filter =/